workflow/snakemake_rules/export_for_nextstrain.smk

```diff
@@ -227,34 +228,72 @@ rule trigger_phylogenetic_rebuild:
         data = json.dumps(
             {"event_type": dispatch_type}
         )
+        send_slack_message(f"This pipeline is now triggering a GitHub action (via dispatch type {dispatch_type}) to run the corresponding phylogenetic builds.")
         print(f"Triggering ncov rebuild GitHub action via repository dispatch type: {dispatch_type}")
         response = requests.post("https://api.github.com/repos/nextstrain/ncov/dispatches", headers=headers, data=data)
         response.raise_for_status()
 
+storage = PersistentDict("slack")
+
+def send_slack_message(message, broadcast=False):
+    """
+    Sends slack messages notifying us of the pipeline's progress. Messages will be
+    threaded, with the first message being the parent message. Important messages can be
+    broadcast: they will be part of the thread but also sent to the channel.
+    """
+    ## Note: this cannot currently send files, but this would be easy to add.
+    ## Slack docs: https://api.slack.com/methods/files.upload
+
+    if not config.get("slack_token", None) or not config.get("slack_channel", None):
+        print("Cannot send slack message as the config does not define a channel and/or token.")
+        return
+
+    headers = {
+        'Content-type': 'application/json',
+        'authorization': f"Bearer {config['slack_token']}",
+        'Accept': 'application/vnd.github.v3+json'
+    }
+    data = {
+        "channel": config["slack_channel"],
+        "text": message,
+    }
+
+    # if slack_thread_ts has been stored, then there is a parent message, so we thread this messaege
+    try:
+        data["thread_ts"]=str(storage.fetch("slack_thread_ts"))
+        if broadcast:
+            data["reply_broadcast"]=True
+    except NoSuchEntryError:
+        pass
+
+    response = requests.post("https://slack.com/api/chat.postMessage", headers=headers, data=json.dumps(data))
+    response.raise_for_status()

```

Comment:
Yeah. Our daily builds are not critical -- they can fail for a few days without any major side-effects, and I'm gambling that slack will sort itself out in less time than matters for our purposes. As you outline, if we don't explicitly set a slack channel & token these steps are skipped, which would allow us to deploy builds from our local machines.